ansible_base/authentication/serializers/authenticator_map.py

```diff
@@ -17,17 +25,69 @@ def validate(self, data) -> dict:
         map_type = data.get('map_type', None)
         team = data.get('team', None)
         org = data.get('organization', None)
-        if map_type == 'team' and (not team or team == ''):
-            errors["team"] = "You must specify a team with the selected map type"
-        if map_type == 'team' and (not org or org == ''):
-            errors["organization"] = "You must specify an organization with the selected map type"
-        if map_type == 'organization' and (not org or org == ''):
-            errors["organization"] = "You must specify an organization with the selected map type"
+        role = data.get('role', None)
+
+        if map_type == 'team' and not team:
+            errors["team"] = _("You must specify a team with the selected map type")
+        if map_type in ['team', 'organization'] and not org:
+            errors["organization"] = _("You must specify an organization with the selected map type")
+        if map_type in ['team', 'organization', 'role'] and not role:
+            errors["role"] = _("You must specify a role with the selected map type")
+        if map_type in ['allow', 'is_superuser'] and role:
+            errors["role"] = _("You cannot specify role with the selected map type")
+
+        if role:
+            errors.update(self.validate_role_data(map_type, role, org, team))
 
         if errors:
             raise ValidationError(errors)
         return data
 
+    def validate_role_data(self, map_type, role, org, team):
+        errors = {}
+
+        # Validation is possible only if RBAC is installed
+        if not apps.is_installed('ansible_base.rbac'):
+            logger.warning(_("You specified a role without RBAC installed "))
+            return errors
+
+        from ansible_base.rbac.models import RoleDefinition
+
+        try:
+            rbac_role = RoleDefinition.objects.get(name=role)
+            is_system_role = rbac_role.content_type is None
+
+            # system role is allowed for map type == role without further conditions
+            if is_system_role and map_type == 'role':
+                return errors
```

Comment:
Don't you have to add something to `errors` before returning it? It's `{}` right here, as far as I can tell.

Reply:
This is ok. On line 40 we are taking the return from this function and updating the parents errors condition which would then get displayed (if there are any).